docs/architecture/airbyte-specification.md

```diff
@@ -0,0 +1,180 @@
+        * e.g. If we have a table called `users` which had the columns `name` and `age` (the age column is optional) the `AirbyteCatalog` would look like this:
+            ```
+            {
+              "streams": [
+                {
+                  "name": "users",
+                  "schema": {
+                    "type": "object",
+                    "required": ["name"],
+                    "properties": {
+                      "name": {
+                        "type": "string"
+                      },
+                      "age": {
+                        "type": {
+                          "number"
+                        }
+                      }
+                    }
+                  }
+                }
+              ]
+            }
+            ```
+    * If you are using a data source that wraps an API with multiple different resources (e.g. `api/customers` and `api/products`) each route would correspond to a stream. The json object returned by each route would be described in the `json_schema` field.
+        * e.g. In the case where the API has two endpoints `api/customers` and `api/producs` and each returns a list of json objects, the `AirbyteCatalog` might look like this. (Note: using the json schema standard for defining a stream allows you to describe nested objects. You are not constrained to a classic "table/columns" structure)
+            ```
+            {
+              "streams": [
+                {
+                  "name": "customers",
+                  "schema": {
+                    "type": "object",
+                    "required": ["name"],
+                    "properties": {
+                      "name": {
+                        "type": "string"
+                      }
+                    }
+                  }
+                },
+                {
+                  "name": "products",
+                  "schema": {
+                    "type": "object",
+                    "required": ["name", "features"],
+                    "properties": {
+                      "name": {
+                        "type": "string"
+                      },
+                      "features": {
+                        "type": "array",
+                        "items": {
+                          "type": "object",
+                          "required": ["name", "productId"],
+                          "properties": {
+                            "name": "string",
+                            "productId": "number"
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              ]
+            }
+            ```
+#### Read
+* Input:
+    1. config: A configuration json object that has been validated using the `ConnectorSpecification`.
+    1. catalog: An `AirbyteCatalog`. This catalog should be a subset of the catalog returned by the `discover` command.
```

Comment:
Can you add something about how this catalog is also what's used to determine what to select?